### PR TITLE
Add support for drag and drop file uploads

### LINKED ISSUE #38: Drag to attach file
I really miss the ability to drag a file to the input area to attach it. 
I use feature this a lot in Goofy, the client I currently use.

### PR DESCRIPTION
**Pull request by peterjosling**
Will require manually merging after other PRs.

As implemented in Goofy, https://github.com/danielbuechele/goofy/commit/2fa51cdd2e5379a7159d2195cbc05048020e794f

Closes #38

### CODE REVIEW

**Comment by harkmall** [comment]
Any way to make it work like in chat on Facebook.com instead of auto-sending the file?

**Comment by peterjosling** [comment]
Messenger.com doesn't provide any way to preview before sending, so would have to build some UI for it. Could read in the file data, display it and have a confirm button fairly easily, though. Would just be a question of whether or not it's worth it for a feature that will probably be included in the official application fairly soon. (I'm surprised they shipped without it).

File: Messenger/AppDelegate.mm [hunk 1]
```diff
@@ -304,6 +304,8 @@ - (void)webView:(WebView *)webView didFinishLoadForFrame:(WebFrame *)frame {
     NSLog(@"%@%@ frame.dataSource.response=%@", self, NSStringFromSelector(_cmd), frame.dataSource.response);
     [webView.mainFrame.windowObject evaluateWebScript:
      [NSString stringWithFormat:@"document.body.innerText = ''; var e = document.createElement('p'); document.body.appendChild(e); e.innerText = 'Oh noes. It appears Messenger.com is down for maintenance. Please try again later.'; var s = e.style; s.font='18px helvetica-light'; s.lineHeight='27px'; s.color='#999'; s.margin='0 auto'; s.width='50%%'; s.textAlign='center'; s.margin='0 auto'; s.marginTop='100px'; s.marginBottom='30px'; s.width='235px'; s.height='235px'; s.paddingTop='250px'; s.backgroundRepeat='no-repeat'; s.backgroundPosition='top center'; s.backgroundImage='url(%@)';", kErrorPNGDataURL]];
+  } else {
+    [webView.mainFrame.windowObject evaluateWebScript:@"document.addEventListener('dragover', function(e) {e.stopPropagation();e.preventDefault();e.dataTransfer.dropEffect = 'copy';});document.addEventListener('drop', function(e) {e.stopPropagation();e.preventDefault();document.getElementById('js_2').files = e.dataTransfer.files;});"];
```
Comment (rsms) [inline Messenger/AppDelegate.mm:308]: Refrain from referring to elements by class name or ID as the DOM is automatically generated; class names and IDs can change at any time. Instead, use querySelector, i.e:

``` js
document.querySelector('input[type="file"][name="attachment[]"]');
```

**Comment by rsms** [comment]
@golfguru1 @peterjosling All Messenger products send attachments separately from messages. It's a design property of the product rather than a technical limitation or issue. Changing the behaviour of Messenger is unacceptable IMHO. This project aims to provide the Messenger experience on desktop, not a "Messenger-like" experience.

**Pull request merged**